tests/cnf/ran/powermanagement/tests/cpufreq.go

```diff
@@ -0,0 +1,83 @@
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/powermanagement/internal/tsparams"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"k8s.io/utils/cpuset"
+)
+
+var _ = Describe("CPU frequency tuning tests", Label(tsparams.LabelCPUFrequency), func() {
+	var (
+		perfProfile             *nto.Builder
+		desiredReservedCoreFreq = performancev2.CPUfrequency(2500001)
+		desiredIsolatedCoreFreq = performancev2.CPUfrequency(2200001)
+		originalIsolatedCPUFreq performancev2.CPUfrequency
+		originalReservedCPUFreq performancev2.CPUfrequency
+	)
+
+	BeforeEach(func() {
+		var err error
+
+		perfProfile, err = helper.GetPerformanceProfileWithCPUSet()
+		Expect(err).ToNot(HaveOccurred(), "Failed to get performance profile")
+
+		// Get isolated core ID
+		isolatedCPUSet, err := cpuset.Parse(string(*perfProfile.Object.Spec.CPU.Isolated))
+		Expect(err).ToNot(HaveOccurred(), "Failed to get isolated cpu set")
+		isolatedCPUsList := isolatedCPUSet.List()
+		isolatedCPUNumber := isolatedCPUsList[0]
+
+		// Get reserved core ID
+		reservedCPUSet, err := cpuset.Parse(string(*perfProfile.Object.Spec.CPU.Reserved))
+		Expect(err).ToNot(HaveOccurred(), "Failed to get reserved cpu set")
+		reservedCPUsList := reservedCPUSet.List()
+		reservedCPUNumber := reservedCPUsList[0]
+
+		By("get original isolated core frequency")
+		spokeCommand := fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+			isolatedCPUNumber)
+		consoleOut, err := cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+		Expect(err).ToNot(HaveOccurred(), "Failed to %s, error:%s", spokeCommand, consoleOut)
+		freqAsInt, err := strconv.Atoi(strings.TrimSpace(consoleOut))
+		Expect(err).ToNot(HaveOccurred(), "strconv.Atoi Failed")
+		originalIsolatedCPUFreq = performancev2.CPUfrequency(freqAsInt)
+
+		By("get original reserved core frequency")
+		spokeCommand = fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+			reservedCPUNumber)
+		consoleOut, err = cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+		Expect(err).ToNot(HaveOccurred(), "Failed to %s, error: %s", spokeCommand, consoleOut)
+		freqAsInt, err = strconv.Atoi(strings.TrimSpace(consoleOut))
+		Expect(err).ToNot(HaveOccurred(), "strconv.Atoi Failed")
+		originalReservedCPUFreq = performancev2.CPUfrequency(freqAsInt)
+
+	})
+
+	AfterEach(func() {
+		By("Reverts the CPU frequencies to the original setting")
+		err := helper.SetCPUFreq(perfProfile,
+			&originalIsolatedCPUFreq, &originalReservedCPUFreq)
+		Expect(err).ToNot(HaveOccurred(), "Failed to set CPU Freq")
+	})
+
+	Context("Reserved Core Frequency Tuning Test", func() {
+
+		It("tests changing reserved and isolated CPU frequencies", func() {
+			By("patch performance profile to set core frequencies")
+			err := helper.SetCPUFreq(perfProfile,
```

Comment:
Same as above, can this be one line?

Reply:
Good idea- thanks. Done.

Reply:
It looks like this is still multiple lines

Reply:
Fixed for real this time?